src/extension.ts

```diff
@@ -8,16 +8,26 @@ import * as path from 'path';
 import * as os from 'os';
 import * as getPort from 'get-port';
 
-import { workspace, ExtensionContext, Disposable, window, Uri, commands, OutputChannel } from 'vscode';
+import { workspace, ExtensionContext, Disposable, window, Uri, commands, OutputChannel, extensions } from 'vscode';
 import { RevealOutputChannelOn, LanguageClientOptions, ErrorHandler, Message, ErrorAction, CloseAction } from 'vscode-languageclient';
 import { LanguageClient, ServerOptions } from 'vscode-languageclient/node';
+import { PythonExtension, PYTHONEXTENSIONID } from './python_extension';
 
 
-function log(message: string) {
+export function log(message: string) {
 	console.log(`[${new Date().toUTCString()}][vscode-inmanta] ${message}`);
 }
 
 export async function activate(context: ExtensionContext) {
```

Comment:
There is a lot I would change about this activate function.
To make it more readable I would extract all declarations outside the function scope.
But since most of this code was already present, there is no need to change it.